### PR TITLE
API_token generated&deactivated through AJAX

### PR DESCRIPTION
**Pull request by chu4**
Added API token generation and deactivation through AJAX without re-rendering the page of profile

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added
- [ ] screenshot of feature is added if task has UI

### CODE REVIEW

File: app/controllers/account/tokens_controller.rb [hunk 1]
```diff
@@ -2,18 +2,23 @@ class Account::TokensController < ApplicationController
...
   def destroy
     current_user.deactivate_token
-    redirect_to account_profile_path,
-                flash: { info: "Your API key was successfully deactivated" }
+    respond_to_format
+  end
+
+  private
+
+  def respond_to_format(block = nil)
```
Comment (romaslmd) [inline app/controllers/account/tokens_controller.rb]: `(&block)`

File: app/controllers/account/tokens_controller.rb [hunk 2]
```diff
@@ -2,18 +2,23 @@ class Account::TokensController < ApplicationController
...
+    respond_to { |format| format.js { block } }
```
Comment (romaslmd) [inline app/controllers/account/tokens_controller.rb]: `respond_to { |format| format.js(&block) }`

File: app/views/account/tokens/create.js.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+$('#del_token').remove();
+$('#gen_token').hide().after('<%= j render "/account/profile/with_token", profile: @user %>');
```
Comment (romaslmd) [inline app/views/account/tokens/create.js.erb]: mb `replaceWith` ?

Comment (chu4) [reply app/views/account/tokens/create.js.erb]: yep, definitely looks better

**Pull request merged**